daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-cloudevents.md

```diff
@@ -65,6 +65,7 @@ As another example of a v1.0 CloudEvent, the following shows data as XML content
 ## Publish your own CloudEvent
 
 If you want to use your own CloudEvent, make sure to specify the [`datacontenttype`]({{< ref "pubsub-overview.md#setting-message-content-types" >}}) as `application/cloudevents+json`.
+If the CloudEvent that was authored by the app does not contain the minimum required CloudEvent specification, the message will be rejected. If the CloudEvent is valid, Dapr will send it as is and will not add additional fields to it.
```

Comment:
It would be very helpful to document what that minimum required CloudEvent spec is :)

Reply:
It's also my understanding that Dapr will augment with additional properties if they haven't been specified. Which properties are they?

<img width="601" alt="image" src="https://github.com/dapr/docs/assets/4224880/7e05e602-6ae1-4b9d-9e4d-db4d8ffb9182">

Reply:
That's correct, updated with the fields data.

Reply:
@yaron2 awesome!

I notice that `datacontenttype` is optional according to the CE spec, as is `dataschema` `subject` and `time`.

If I were to provide any of `dataschema` `subject` `time`, would dapr pass these through or would it stripped or preserved?

Equally, If I were to append properties to the envelope that were completely outside of the CE spec, would these get stripped or preserved?

Reply:
If you pass any field, part of the CE spec or not, Dapr would pass these through.

Reply:
Got it. Might be worth including that in the docs too?

Second thought that _might_ cause confusion for SDK users where `CloudEvent<T>` may not have an ability to carry arbitrary fields. Hmm.

Reply:
Added a note. Users using the SDK should be mindful of its behavior - from this documentation perspective, the payload is JSON.